### PR TITLE
fix: search top bar alignment and interactions [WPB-2231] [WPB-5172]

### PR DESCRIPTION
**Pull request by saleniuk**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-2231" title="WPB-2231" target="_blank">BugWPB-2231</a>  Longclick on magnifier Icon activates search, but search is not possible
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

We had some issues with SearchTopBar. We fixed clicking on it by covering it with a clickable box, but it made automation fail. There were still issues with text alignment when gaining focus, long-tapping, clickable or non-clickable icons and generally a little bit overcomplicated code of that SearchTopBar with two parameters "active" and "focused".

### Solutions

- moved saving `SearchConversationMessagesState` directly to the ViewModel, no need to keep state of the search bar in two separate places
- simplified SearchTopBar, `isActive` is still there but it affects focus directly and active state changes are observed by listening the focus changes so long-tapping on search bar works exactly like on other text inputs
- added `placeholderAlignment` to be able to animate moving from center to left when gaining focus so the user know better what's happening and why

#### How to Test

Just try to search for conversations, messages in a conversation or users when creating new conversation.

### Attachments (Optional)

https://github.com/wireapp/wire-android-reloaded/assets/30429749/9d66f71e-bc2d-4fd6-9dc4-299c65556157

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1543](https://10.10.124.134/job/AR-build-pipeline/1543/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7934749)(7934749)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7934749)(7934749)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/ui/common/textfield/WireTextField.kt [hunk 1]
```diff
@@ -261,14 +264,18 @@ private fun InnerText(
                 top = 2.dp, bottom = 2.dp
             )
             if (value.text.isEmpty() && placeholderText != null) {
-                Text(
-                    text = placeholderText,
-                    style = placeholderTextStyle,
-                    color = colors.placeholderColor(state).value,
+                Column(
```
Comment (Garzas) [inline app/src/main/kotlin/com/wire/android/ui/common/textfield/WireTextField.kt]: why column?

**Review by Garzas — APPROVED** [review]
Looking good, just one compose question 😄

File: app/src/main/kotlin/com/wire/android/ui/common/textfield/WireTextField.kt (hunk 1 above)
Comment (saleniuk) [reply app/src/main/kotlin/com/wire/android/ui/common/textfield/WireTextField.kt]: I wanted to add parameter to allow only for `Alignment.Horizontal`, but you motivated me to remove this `Column` to have less nesting and create a mapper from `Alignment.Horizontal` to `Alignment` 😄

**Comment by AndroidBob** [comment]
Build [1562](https://10.10.124.134/job/AR-build-pipeline/1562/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7943882)(7943882)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7943882)(7943882)-staging-compat.apk)

**Review by ohassine — APPROVED** [review]

**Pull request merged**